### PR TITLE
docs: Add a how to guide for making new/existing repository in the org

### LINKED ISSUE #25: Add how to start new repository Contributing guide
Add a how-to guide for starting a repository in AUV ZHCET organization.

### PR DESCRIPTION
**Pull request by Zaidtech**
# Checklist

- [x] Closes #25
- [x] Passes `Semantic Pull Request` and `TypoCheck`
- [ ] New functions/methods are listed in `README.md`
- [x] New functions/methods are listed in `index.md`
- [ ] Updated table of content
- [x] Updated nav links in mkdocs.yml

---

## PR description
Solves issue #25 -how to start making a repository guide.

### CODE REVIEW

File: docs/Development/Creating_a_Repository_guide.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# GIT REPOSITORY HOW-TO GUIDE
```
Comment (pnijhara) [inline docs/Development/Creating_a_Repository_guide.md]: Remove the empty space after end of line.

File: docs/Development/Creating_a_Repository_guide.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# GIT REPOSITORY HOW-TO GUIDE 
+### git init/ git clone
```
Comment (pnijhara) [inline docs/Development/Creating_a_Repository_guide.md]: Leave an empty line after line1

File: docs/Development/Creating_a_Repository_guide.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+A *[git repository](https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository)* is a virtual storage of your project, allowing to save versions of your code and to access when needed.
+
+## SETTING UP A REPOSITORY (Repo) UNDER THE GIT VERSION CONTROL
+[How to set-up Git on your device](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup)
```
Comment (pnijhara) [inline docs/Development/Creating_a_Repository_guide.md:9]: Leave a line after the heading.

File: docs/Development/Creating_a_Repository_guide.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+### Initializing a new repository: **git init**
+To create a new repo, you'll use the **git init** command. git init is a one-time command you use during the initial setup of a new repo. Executing this command will create a new **.git**   subdirectory in your current working directory. This will also create a new branch called master.
```
Comment (pnijhara) [inline docs/Development/Creating_a_Repository_guide.md]: Headings should be surrounded by blank lines.

File: docs/Development/Creating_a_Repository_guide.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+### Versioning an existing project with a new git repository
+Assuming you already have a project directory and you would like to create a repo within it:
+
+* First cd into root project directory.
+* execute **git init** 
+This will as mentioned above create same **.git** subdirectory but scoped to that project dir.
+
+### Cloning an existing repo
+If a project has been set up in a central repository, the **clone** command is the most common way to obtain a local development clone.Once a developer(contributor) has obtained a working copy, all version control operations are managed through their local repository.
```
Comment (pnijhara) [inline docs/Development/Creating_a_Repository_guide.md:25]: Headings should be surrounded by blank lines.

File: docs/Development/Creating_a_Repository_guide.md (hunk 4 above)
Comment (pnijhara) [inline docs/Development/Creating_a_Repository_guide.md]: Remove empty space after the end of line.

File: docs/Development/Creating_a_Repository_guide.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+[How to set-up Git on your device](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup)
+
+### Initializing a new repository: **git init**
+To create a new repo, you'll use the **git init** command. git init is a one-time command you use during the initial setup of a new repo. Executing this command will create a new **.git**   subdirectory in your current working directory. This will also create a new branch called master. 
+
+### Versioning an existing project with a new git repository
+Assuming you already have a project directory and you would like to create a repo within it:
+
+* First cd into root project directory.
+* execute **git init**
```
Comment (pnijhara) [inline docs/Development/Creating_a_Repository_guide.md]: Remove empty space.

File: docs/Development/Creating_a_Repository_guide.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+Assuming you already have a project directory and you would like to create a repo within it:
+
+* First cd into root project directory.
+* execute **git init** 
+This will as mentioned above create same **.git** subdirectory but scoped to that project dir.
+
+### Cloning an existing repo
+If a project has been set up in a central repository, the **clone** command is the most common way to obtain a local development clone.Once a developer(contributor) has obtained a working copy, all version control operations are managed through their local repository.
+    
+    git clone <repo url>
```
Comment (pnijhara) [inline docs/Development/Creating_a_Repository_guide.md]: No need to add indentation.
This should be "a code block".

File: docs/Development/Creating_a_Repository_guide.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    
+ git clone creates a local clone of remote repo.
```
Comment (pnijhara) [inline docs/Development/Creating_a_Repository_guide.md]: Remove empty space.

File: docs/Development/Creating_a_Repository_guide.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+This will as mentioned above create same **.git** subdirectory but scoped to that project dir.
+
+### Cloning an existing repo
+If a project has been set up in a central repository, the **clone** command is the most common way to obtain a local development clone.Once a developer(contributor) has obtained a working copy, all version control operations are managed through their local repository.
+    
+    git clone <repo url>
+    
+ git clone creates a local clone of remote repo. 
+ 
+ The URL can be found on this page under **CODE** dropdown can be found [on this page.](https://github.com/auvzhcet/Documentation)
```
Comment (pnijhara) [inline docs/Development/Creating_a_Repository_guide.md]: Remove empty space before starting of the line.

File: docs/Development/Creating_a_Repository_guide.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+ 
+ ### Saving changes to the repo
```
Comment (pnijhara) [inline docs/Development/Creating_a_Repository_guide.md]: Headings should be surrounded by blank lines.

File: docs/Development/Creating_a_Repository_guide.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+ 1. add changes to staging area    
+     
+       git add <file names or the whole dir(./ )>
+ 
+ 2. Commit those changes
+ 
+        git commit -m "Message about the commit"
+
+### Collaboration
+**git push**
```
Comment (pnijhara) [inline docs/Development/Creating_a_Repository_guide.md]: Headings should be surrounded by blank lines.

**Review by pnijhara — CHANGES_REQUESTED** [review]

**Comment by pnijhara** [comment]
@Zaidtech There are linting issues in this patch. I have suggested some. If you are using VScode I suggest you install markdown linter extension in your code editor.

**Comment by Zaidtech** [comment]
@pnijhara I have resolved the issue, kindly check them.

File: .idea/.gitignore [hunk 12]
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
```
Comment (pnijhara) [inline .idea/.gitignore]: Remove these files.

File: .idea/Documentation.iml [hunk 13]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (pnijhara) [inline .idea/Documentation.iml]: Remove this file.

File: .idea/Documentation.iml [hunk 14]
```diff
@@ -0,0 +1,8 @@
...
+<module type="PYTHON_MODULE" version="4">
+  <component name="NewModuleRootManager">
```
Comment (pnijhara) [inline .idea/Documentation.iml]: Remove this file.

File: .idea/inspectionProfiles/profiles_settings.xml [hunk 15]
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
+  <settings>
+    <option name="USE_PROJECT_PROFILE" value="false" />
```
Comment (pnijhara) [inline .idea/inspectionProfiles/profiles_settings.xml]: Remove this file.

File: .idea/misc.xml [hunk 16]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="JavaScriptSettings">
```
Comment (pnijhara) [inline .idea/misc.xml]: Remove this file.

File: .idea/modules.xml [hunk 17]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (pnijhara) [inline .idea/modules.xml]: Remove this file.

File: .idea/vcs.xml [hunk 18]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (pnijhara) [inline .idea/vcs.xml]: Remove this file.

**Review by pnijhara — CHANGES_REQUESTED** [review]

File: docs/Development/Creating_a_Repository_guide.md (hunk 4 above)
Comment (pnijhara) [reply docs/Development/Creating_a_Repository_guide.md]: Add a line after heading.

**Comment by Zaidtech** [comment]
Removed the .idea files. @pnijhara, please check the same.

**Review by pnijhara — APPROVED** [review]

File: docs/Development/Creating_a_Repository_guide.md [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+
+A *[git repository](https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository)* is a virtual storage of your project, allowing to save versions of your code and to access when needed.
+
+## SETTING UP A REPOSITORY (Repo) UNDER THE GIT VERSION CONTROL
+
+[How to set-up Git on your device](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup)
+
+### Initializing a new repository: **git init**
+
+To create a new repo, you'll use the **git init** command. git init is a one-time command you use during the initial setup of a new repo. Executing this command will create a new **.git**   subdirectory in your current working directory. This will also create a new branch called master.
```
Comment (syedjawadakhtar) [inline docs/Development/Creating_a_Repository_guide.md]: When using terminal commands in content, better show it between three back quotes (`)
Like this:
```zsh
git init
```

**Review by syedjawadakhtar — CHANGES_REQUESTED** [review]
This article must be more focused on the format of the title of the repository, what to write in README, and how to properly document so that the code could be replicated by anyone.
Following the basic structure from here: [Launching your own open-source project - opensource.guide](https://opensource.guide/starting-a-project/#launching-your-own-open-source-project) - providing the sample files to each one of these.

These git commands can be added in a subsection as "Git Command Guide".

**Comment by pnijhara** [comment]
For starting I think this much content is fine. We will update this with time. I am merging this one for now.

**Review by pnijhara — APPROVED** [review]

**Comment by pnijhara** [comment]
Thank you for the contribution @Zaidtech

**Pull request merged**